src/comparator/src/ArtifactDelta.ts

```diff
@@ -57,6 +57,14 @@ export default class ArtifactDelta {
     return this._hashChanged;
   }
 
+  public get hashChangeUnexpected(): boolean {
+    return (
+      this._hashChanged &&
+      this.failingBudgets.length === 0 &&
+      Object.keys(this.sizes).every(sizeKey => this.sizes[sizeKey] === 0)
```

Comment:
I guess there's a small chance this could flag some false positives, where the contents changed and therefore the hash changed, but the size (coincidentally) stayed exactly the same. It's a bit of an edge case, but if we did want to fix it, we could do something like

```ts
const providedHashHasChanged = prevProvidedHash !== providedHash;
const calculatedHashHasChanged = calculateHashFromContents(prevContent) !== calculateHashFromContents(content);
const hashChangedUnexpected = providedHashHasChanged && (calculatedHashHasChanged === false);
```

(where "provided hash" is the one returned by the user's `getFilenameHash`)

Reply:
We don't store the calculated hash if a hash is provided. I am afraid this also overcomplicates things a bit and am hesitant to go down this road.